src/constructs/static-site/static-site.ts

```diff
@@ -0,0 +1,322 @@
+   @see https://docs.aws.amazon.com/cdk/v2/guide/tokens.html
+    */
+  private static getResourceNameFromArn(certificateArn: string): string {
+    const { account, region, resource, resourceName } = Arn.split(certificateArn, ArnFormat.SLASH_RESOURCE_NAME);
+
+    if (!Token.isUnresolved(account)) {
+      throw new Error(
+        "Account numbers are considered private information and should not be added to VCS. Use the `account` property on `GuStack` instead."
+      );
+    }
+
+    if (region !== GLOBAL_AWS_REGION) {
+      throw new Error(CLOUDFRONT_CERTIFICATE_REQUIREMENT_MESSAGE);
+    }
+
+    if (resource !== ACM_ARN_RESOURCE || !resourceName) {
+      throw new Error("Invalid ARN for an ACM resource.");
+    }
+
+    return resourceName;
+  }
+
+  /*
+  CloudFront requires an ACM certificate from us-east-1 (the "global" region).
+  If a certificateArn has been provided in the props, validate it is from the global region, or throw.
+  Else, create a new certificate and show a reminder to deploy the template in the global region.
+
+  Once https://github.com/aws-cloudformation/cloudformation-coverage-roadmap/issues/523 ships, this dance should no longer be necessary.
+   */
+  private static getCertificate(
+    scope: GuStack,
+    { app, domainNameProps, preExistingCertificateArn }: GuStaticSiteProps
+  ) {
+    if (!preExistingCertificateArn) {
+      Annotations.of(scope).addInfo(
+        `Deploy this template in the ${GLOBAL_AWS_REGION} region as ${CLOUDFRONT_CERTIFICATE_REQUIREMENT_MESSAGE}`
+      );
+
+      return new GuCertificate(scope, {
+        app,
+        ...domainNameProps,
+      });
+    }
+
+    const resourceName = StageAwareValue.isStageValue(preExistingCertificateArn)
+      ? scope.withStageDependentValue({
+          app,
+          variableName: "certificateResourceName",
+          stageValues: {
+            [Stage.CODE]: GuStaticSite.getResourceNameFromArn(preExistingCertificateArn.CODE),
+            [Stage.PROD]: GuStaticSite.getResourceNameFromArn(preExistingCertificateArn.PROD),
+          },
+        })
+      : GuStaticSite.getResourceNameFromArn(preExistingCertificateArn.INFRA);
+
+    return Certificate.fromCertificateArn(
+      scope,
+      AppIdentity.suffixText({ app }, "Certificate"),
+      Arn.format({
+        partition: "aws",
+        service: "acm",
+        region: GLOBAL_AWS_REGION,
+        account: scope.account,
+        resource: ACM_ARN_RESOURCE,
+        resourceName,
+        arnFormat: ArnFormat.SLASH_RESOURCE_NAME,
+      })
+    );
+  }
+
+  private static getBucket(
+    scope: GuStack,
+    { app, preExistingOriginBucketName, bucketName }: GuStaticSiteProps,
+    { cloudFrontOriginAccessIdentityS3CanonicalUserId }: OriginAccessIdentity
+  ) {
+    const bucket = preExistingOriginBucketName
+      ? Bucket.fromBucketName(scope, AppIdentity.suffixText({ app }, "OriginBucket"), preExistingOriginBucketName)
+      : new GuS3Bucket(scope, "OriginBucket", { app, bucketName });
```

Comment:
I think this will create a bucket per stage? But in the PR description you mentioned that:

> Riff-Raff's aws-s3 type can only upload to a single bucket and will shard the bucket by stage. 

I'm wondering if we should force users to pass in a pre-existing bucket unless/until we change Riff-Raff?

Reply:
Alternatively we could use a [condition](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/conditions-section-structure.html) to create the bucket in one stage only, but that feels pretty gross!

Reply:
Ah, hadn't noticed this, great spot! Conditions are pretty horrible imo and I'd prefer to keep this kind of logic in the domain of TypeScript (i.e. compile time).

Maybe an alternative approach could be to require an [`IBucket`](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-s3.IBucket.html) and leave it to the user to either create a new bucket or use `Bucket.fromBucketName`?

Reply:
For further discussion - https://github.com/guardian/recommendations/discussions/74.